### PR TITLE
[iOS] Enable Diagnostics.Tracing tests

### LINKED ISSUE #51382: System.Diagnostics.Tracing.Tests Fails on iOS and tvOS
System.Diagnostics.Tracing.Tests
BasicEventSourceTests.TestsManifestNegative.Test_GenerateManifest_InvalidEventSources
BasicEventSourceTests.TestsTraits.Test_EventSource_Traits_Dynamic
BasicEventSourceTests.TestsTraits.Test_EventSource_Traits_Contract
BasicEventSourceTests.TestsWriteEventToListener.Test_EventSourceCreatedEvents_BeforeListener
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ArgsBasicTypes
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_InvalidCalls
BasicEventSourceTests.TestsWriteEventToListener.Test_EventSourceCreatedEvents_AfterListener
BasicEventSourceTests.TestsWriteEventToListener.Test_EventListenerThrows_ExceptionIsNotRethrownToCaller
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ArgsCornerCases
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ZeroKwds
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ToChannel_Coverage
BasicEventSourceTests.TestsWrite.Test_Write_T_EventListener
BasicEventSourceTests.TestsWrite.Test_Write_T_EventListener_UseEvents
BasicEventSourceTests.TestsUserErrors.Test_BadEventSource_MismatchedIds

```
System.Diagnostics.Tracing.Tests.dll   Failed: 13

Test collection for System.Diagnostics.Tracing.Tests.dll
BasicEventSourceTests.TestsManifestNegative.Test_GenerateManifest_InvalidEventSources
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsTraits.Test_EventSource_Traits_Dynamic
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsTraits.Test_EventSource_Traits_Contract
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_EventSourceCreatedEvents_BeforeListener
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ArgsBasicTypes
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_InvalidCalls
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_EventSourceCreatedEvents_AfterListener
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ArgsCornerCases
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ZeroKwds
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWriteEventToListener.Test_WriteEvent_ToChannel_Coverage
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWrite.Test_Write_T_EventListener
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsWrite.Test_Write_T_EventListener_UseEvents
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)
BasicEventSourceTests.TestsUserErrors.Test_BadEventSource_MismatchedIds
    Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: \nActual:   Private.InternalDiagnostics.System.Net.So···\n          ↑ (pos 0)

### PR DESCRIPTION
**Pull request by steveisok**
Fixes an issue where a test project that contains native libraries will fool the AOT compiler task in thinking there are assemblies in different locations. This causes .aotdata files to not be bundled with the test app and ends up causing a runtime assertion.

Fixes https://github.com/dotnet/runtime/issues/51382

### CODE REVIEW

**Comment by steveisok** [comment]
@radical I'm not sure if switching the Filter / Ensure methods are going to impact you.

**Comment by radical** [comment]
> @radical I'm not sure if switching the Filter / Ensure methods are going to impact you.

I think it should be fine. `FilterAssemblies` does not try to load any dependencies, so location shouldn't matter. But let's see what the tests say :)

**Review by radical — APPROVED** [review]

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -311,7 +311,6 @@
 
     <!-- App Crash https://github.com/dotnet/runtime/issues/53624 -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj" />
-    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Diagnostics.Tracing/tests/System.Diagnostics.Tracing.Tests.csproj" />
```
Comment (mdh1418) [inline src/libraries/tests.proj:314]: We also need to remove the ActiveIssues like https://github.com/dotnet/runtime/blob/bdc0d1a409a24d4092e7b108c13fc4ff1c643d1f/src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestNegative.cs#L62

**Review by mdh1418 — CHANGES_REQUESTED** [review]

**Comment by mdh1418** [comment]
We can reenable these as well https://github.com/dotnet/runtime/issues/56073
We just need to add ```RuntimeComponents="$(RuntimeComponents)"``` to `AppleAppBuilderTask` in `AppleApp.targets` and setting this property in `System.Diagnostics.Tracing.Tests.csproj` ```<RuntimeComponents>diagnostics_tracing</RuntimeComponents>```

**Comment by radical** [comment]
```
Precompiling failed for /datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll with exit code 1.
Failed to load method 0x6000065 from '/datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll' due to Could not load file or assembly 'Microsoft.JSInterop.WebAssembly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies..
Run with MONO_LOG_LEVEL=debug for more information.
AOT of image Microsoft.AspNetCore.Components.WebAssembly.dll failed.
Mono Ahead of Time compiler - compiling assembly /datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll
```

Hm `Failed to load method 0x6000065 from '/datadisks/disk1/work/A48F08FA/w/A9BB08F7/e/blazorwasm_Release_aot/obj/Release/net7.0/wasm/for-publish/aot-in/Microsoft.AspNetCore.Components.WebAssembly.dll' due to Could not load file or assembly 'Microsoft.JSInterop.WebAssembly, ..`

`Microsoft.JSInterop.WebAssembly` is skipped for AOT, but looks like it is still needed by `mono-aot-cross`. Hm I'll push a fix for this.

**Comment by radical** [comment]
> Fixes an issue where a test project that contains native libraries will fool the AOT compiler task in thinking there are assemblies in different locations. This causes .aotdata files to not be bundled with the test app and ends up causing a runtime assertion.

Could you explain what exactly is happening here?
- test project contains native dlls, in location different from the managed assemblies
- which makes the aot compiler task to move all of them (including the native ones) to a single directory
- but how does that affect the `aotdata` being bundled?

**Comment by radical** [comment]
I have pushed some changes that fixes the issue in two ways:

1. The `.aotdata` files will be emitted next to the original assemblies
2. Unmanaged assemblies won't be used to decide whether we need to move all the assemblies to a `aot-in` directory

**Comment by steveisok** [comment]
System.Diagnostics.Tracing tests pass on tvOS.  The other failures are unrelated.

File: src/libraries/tests.proj [hunk 2]
```diff
@@ -278,6 +278,19 @@
...
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Formats.Cbor\tests\System.Formats.Cbor.Tests.csproj" />
+
+    <!--
+      Test apps that are too large and take too long to build 
+      Keep here until aggressive trimming targets can work on helix.
+    -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.RegularExpressions\tests\UnitTests\System.Text.RegularExpressions.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Tests.csproj" />
+
+
```
Comment (mdh1418) [inline src/libraries/tests.proj:293]: Are these new failures? Are there issues associated that we can put in the comments?

**Review by mdh1418 — APPROVED** [review]
Looks good to me! If we can add new issues/reference existing ones for any new project/test skips that'd be good.

File: src/libraries/tests.proj (hunk 2 above)
Comment (radical) [reply src/libraries/tests.proj:293]: If it's just links that we need to add, then we can do that in a follow up PR.

Comment (steveisok) [reply src/libraries/tests.proj:293]: There will be.  I would rather address it in https://github.com/dotnet/runtime/pull/67861

**Pull request merged**